framework/doc/content/source/meshgenerators/CoarsenBlockGenerator.md

```diff
@@ -0,0 +1,49 @@
+# CoarsenBlockGenerator
+
+!syntax description /Mesh/CoarsenBlockGenerator
+
+## Overview
+
+This MeshGenerator object allows the user to coarsen one or more blocks in a mesh. It has been designed
+to work with previously refined meshes that were saved to a file without the mesh refinement pattern information. For example, refined meshes stored using [Exodus.md] or [VTK.md] file formats do not contain the information needed to recognize parent and children elements from uniform refinement. The heuristic can still work in the general case of non-uniform refinement.
+
+!alert note
+If elements are refined during the mesh generation process, this generator will not be able to coarsen
+them.
+
+The user has to provide the ids of the blocks to be modified, as well as the corresponding levels of coarsening for each block. These must match up to the order of the aforementioned block id list, e.g. if the blocks you wish to modify are '0 2 1 4', with 1 coarsening for block 0, 2 for block 2, 3 for block 1, and 4 for block 4, then the coarsening list will need to look like '1 2 3 4'.
+
+The user provides the location of starting point for the coarsening algorithm with the [!param](/Mesh/CoarsenBlockGenerator/starting_point) parameter. The element containing that point is found, then the algorithm attempts to coarsen by selecting each node of that element as the interior node of a coarse element containing that element. Once this element is coarsened, the neighbors of the coarse element
+are considered as targets for the next round of coarsening, until all elements are coarsened once for that round of coarsening. If
```

Comment:
How does this work with the block restriction? E.g. what happens if the starting point is in a subdomain that is surrounded by blocks not specified for coarsening, but there are other blocks specified for coarsening? I looked and saw that it is not a `std::vector<Point>` for the `starting_point` parameter, but just a single `Point`

Reply:
if there is no contiguous path to reaching those blocks from the starting_point, you are out of luck.

A vector of starting points would be fine to me

Reply:
It's an evolution compared to what we have now. I had in mind trying to detect automatically where to start instead, in future works

Reply:
Actually this wont work.,

If you pick a single starting point, there are only 4 choices for the central nodes of the coarse elements.
If you pick multiple starting points, it's 4x the choices for every other starting point.